test/metabase/api/action_test.clj

```diff
@@ -53,57 +53,75 @@
               (is (schema= ExpectedGetCardActionAPIResponse
                            action)))))))))
 
-(defn- mock-requests []
-  [{:action       "action/row/create"
-    :request-body (assoc (mt/mbql-query categories) :create-row {:name "created_row"})
-    :expect-fn    (fn [result]
-                    ;; check that we return the entire row:
-                    (is (= "created_row" (get-in result [:created-row :name])))
-                    (is (= (set [:name :id])
-                           (set (keys (:created-row result))))))}
-   {:action       "action/row/update"
-    :request-body (assoc (mt/mbql-query categories {:filter [:= $id 1]})
-                         :update_row {:name "updated_row"})
-    :expected     {:rows-updated [1]}}
-   {:action       "action/row/delete"
-    :request-body (mt/mbql-query categories {:filter [:= $id 1]})
-    :expected     {:rows-deleted [1]}}
-   {:action       "action/row/update"
-    :request-body (assoc (mt/mbql-query categories {:filter [:= $id 10]})
-                         :update_row {:name "new-category-name"})
-    :expected     {:rows-updated [1]}}])
+(defn- format-field-name
+  "Format `field-name` appropriately for the current driver (e.g. uppercase it if we're testing against H2)."
+  [field-name]
+  (keyword (mt/format-name (name field-name))))
+
+(defn- categories-row-count []
+  (first (mt/first-row (mt/run-mbql-query categories {:aggregation [[:count]]}))))
+
+(deftest create-test
```

Comment:
I split out/replaced `happy-path-test` and `create-update-delete-test` with separate `create-test`, `update-test`, and `delete-test`s. They're testing the same thing but it's a lot easier to read them now.

Reply:
It also makes test failures easier to debug since they're more granular -- a bug in `create` won't cause `update` to fail